incydr/utils.py

```diff
@@ -1,40 +1,170 @@
+
+    Will automatically attempt to "render" the field values in the following order:
+       - if `render` arg is a string, it will look in the "extra" section of the pydantic model's Field Info for that
+          name (expects a callable to be there)
+       - if value is of a type that the model has a `json_encoder` for, it will use that encoder
+       - otherwise will leave the value unchanged
+    """
+    fields = get_fields(model.__class__, include=include, flat=flat)
+    for name in fields:
+        path = name.split(".")
+        value, field = model_field_getter(model, path)
+        field_renderer = None if not field else field.field_info.extra.get(render)
+        if render and field_renderer:
+            value = field_renderer(value)
+            yield name, value
+            continue
+        json_encoder = model.Config.json_encoders.get(type(value))
+        if json_encoder:
+            value = json_encoder(value)
+        yield name, value
+
+
+def list_as_panel(items, sep=None, title=None):
+    """
+    Renders a list of items as a `rich.panel.Panel`. If `sep` is provided will add a separator between each item (useful
+    for rendering list of models that have multiple values.
+    """
+    if sep:
+        items = list(chain(*zip(items, repeat(sep))))[:-1]
+    return Panel(Group(*items), title=title)
+
+
+@group()
+def model_as_card(model):
+    """
+    Renders a pydantic model in 'card' format, where field name/value pairs are presented vertically, and when a field
+    is a list of items, it renders it as a separate panel.
+    """
+    for name, value in iter_model_formatted(model, flat=True, render="table"):
+        if isinstance(value, list):
+            if not len(value):
+                yield f"{name}: []"
+            elif isinstance(value[0], BaseModel):
+                yield list_as_panel(
+                    [model_as_card(v) for v in value], title=name, sep="---"
+                )
+            else:
+                yield list_as_panel(value, title=name)
+        elif isinstance(value, ConsoleRenderable):
+            yield Panel.fit(value, title=name)
         else:
-            items.append((new_key, v))
-    return dict(items)
+            yield Text(f"{name}: {value}", overflow="fold")
+
+
+def flatten_fields(model: Type[BaseModel]) -> Generator[str, None, None]:
+    """
+    Yields all fields of a model and any sub-models as flat, dot-notation strings.
+
+    For example, given the following model hierarchy:
+
+        class Child(BaseModel):
+            field_1: str
+            field_2: int
+
+        class Parent(BaseModel):
+            field: str
+            child: Child
+
+    flatten_fields(Parent) would yield: ['field', 'child.field_1', 'child.field_2']
+    """
+    for name, field in model.__fields__.items():
+        if issubclass(field.type_, BaseModel) and field.shape == 1:
```

Comment:
It would be nice to have some in line comments for some of the logic, particularly in these `utils` file methods imo